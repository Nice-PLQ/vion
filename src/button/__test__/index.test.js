import { mount } from '@vue/test-utils';
import Button from '../button.vue';

describe('Button', () => {
  const wrapper = mount(Button, {
    slots: {
      default: '按钮'
    }
  });

  test('按钮类型', async () => {
    await wrapper.setProps({ type: 'primary' });
    expect(wrapper.classes('v-button_primary')).toBe(true);
    expect(wrapper.element).toMatchSnapshot();
  });

  test('按钮大小', async () => {
    await wrapper.setProps({ size: 'large' });
    expect(wrapper.classes('v-button_large')).toBe(true);
  });

  test('按钮禁用', async () => {
    await wrapper.setProps({ disabled: true });
    wrapper.trigger('click');
    expect(wrapper.emitted().click).toBeFalsy();
  });

  test('按钮加载态', async () => {
    await wrapper.setProps({ loading: true });
    expect(wrapper.classes('v-button_loading')).toBe(true);
    expect(wrapper.findComponent('.v-loading')).toBeTruthy();
  });
});
