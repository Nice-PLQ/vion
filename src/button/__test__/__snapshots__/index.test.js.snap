// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button 按钮类型 1`] = `
<button
  class="v-button v-button_primary v-button_medium"
>
  <div
    class="v-button__inner"
  >
    <!--v-if-->
    
    按钮
    
  </div>
</button>
`;
